From 35a6b56e47710ea306608f44382b76022e236be0 Mon Sep 17 00:00:00 2001
From: Matt Peters <matt.peters@windriver.com>
Date: Fri, 25 Oct 2019 11:20:37 -0500
Subject: [PATCH] update init container env to include node name

Update the init containers to have access to the same set of
environment variables as the runtime containers.  Specifically
NODE_NAME is required to ensure the name parameter is set
properly.
---
 stable/filebeat/templates/daemonset.yaml   | 8 ++++++++
 stable/metricbeat/templates/daemonset.yaml | 8 ++++++++
 2 files changed, 16 insertions(+)

diff --git a/stable/filebeat/templates/daemonset.yaml b/stable/filebeat/templates/daemonset.yaml
index 22b46df..2429892 100644
--- a/stable/filebeat/templates/daemonset.yaml
+++ b/stable/filebeat/templates/daemonset.yaml
@@ -54,6 +54,14 @@ spec:
           readOnly: true
           subPath: filebeat.yml
         env:
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
       ## make output.elasticsearch parms visible
       {{- range $key, $value := .Values.config}}
       {{- $configname := $key | upper | replace "." "_" -}}
diff --git a/stable/metricbeat/templates/daemonset.yaml b/stable/metricbeat/templates/daemonset.yaml
index 947e501..247304e 100644
--- a/stable/metricbeat/templates/daemonset.yaml
+++ b/stable/metricbeat/templates/daemonset.yaml
@@ -49,6 +49,14 @@ spec:
           readOnly: true
           subPath: metricbeat.yml
         env:
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
       ## make output.elasticsearch parms visible
       {{- range $key, $value := .Values.daemonset.config}}
       {{- $configname := $key | upper | replace "." "_" -}}
-- 
1.8.3.1

